src/frontend/src/utils/iiConnection.ts

```diff
@@ -473,6 +479,60 @@ export class AuthenticatedConnection extends Connection {
     return await actor.verify_tentative_device(this.userNumber, pin);
   };
 
+  getIdentityInfo = async (): Promise<
+    { Ok: IdentityInfo } | { Err: IdentityInfoError }
+  > => {
+    const actor = await this.getActor();
+    return await actor.identity_info(this.userNumber);
+  };
+
+  setIdentityMetadata = async (
+    metadata: MetadataMapV2
+  ): Promise<{ Ok: null } | { Err: IdentityMetadataReplaceError }> => {
+    const actor = await this.getActor();
+    return await actor.identity_metadata_replace(this.userNumber, metadata);
+  };
+
+  setShownRecoveryWarningPage = async (): Promise<void> => {
```

Comment:
I decided to put the logic of how the timestamp is saved within the `AuthenticatedConnection`. Not a strong opinion, but I didn't find another place.

My idea would be to have different methods to manage different metadata keys. And they might share some logic, but at least have a simple API with a complex implementation.

The downside is that I need to make two calls when setting the metadata.

* To get the metadata
* To replace it.

Reply:
What about switching to a model where the metadata map is loaded (asynchronously) when constructing the `AuthenticatedConnection`, and then modifications (maybe multiple) happen on the cached copy. And then, at some distinct point (in parallel with something that takes a while anyway, e.g. `prepare_delegation`) we write it back?